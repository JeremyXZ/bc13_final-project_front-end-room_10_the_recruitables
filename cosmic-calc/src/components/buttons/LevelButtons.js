import React from "react";
import "./levelButtons.css";

function LevelButtons({ ButtonNumber, clickToGame }) {
  return (
    <div className="levelButton">
      <button
        className={"levelButtons"}
        onClick={clickToGame}
        id={`levelButton-${ButtonNumber}`}
      >
        {/* BUTTON {ButtonNumber} */}
      </button>
    </div>
  );
}

export default LevelButtons;
