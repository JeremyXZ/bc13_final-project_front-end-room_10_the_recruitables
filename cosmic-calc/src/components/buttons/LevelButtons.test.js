import React from "react";
import "@testing-library/jest-dom";
import { screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import LevelButtons from "./LevelButtons";

describe("<LevelButtons/>", () => {
  const mockedProps = {
    ButtonNumber: 7,
    clickToGame: jest.fn(),
    text: "Go to level 7"
  };

  it("successfully renders the component, matches snapshot", () => {
    const { asFragment } = render(<LevelButtons {...mockedProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("click of button fires the provided 'clickToGame' callback prop", () => {
    render(<LevelButtons {...mockedProps} />);
    const button = screen.getByRole("button");
    userEvent.click(button);
    expect(button).toBeInTheDocument();
    expect(mockedProps.clickToGame).toHaveBeenCalled();
  });

  it("button text contains provided 'text' prop content", () => {
    render(<LevelButtons {...mockedProps} />);
    const button = screen.getByRole("button");
   
    // expect(button).toHaveTextContent(mockedProps.text);
    expect(button).toHaveTextContent("Go to level 7");
  });

  it("button attribute contains provided 'buttonNumber' prop content", () => {
    render(<LevelButtons {...mockedProps} />);
    const button = screen.getByRole("button");
    // console.log(button.innerText);
    expect(button).toHaveAttribute("id", "levelButton-7");
  });
});