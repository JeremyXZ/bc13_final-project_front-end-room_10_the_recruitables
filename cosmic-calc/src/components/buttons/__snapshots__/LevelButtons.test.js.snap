// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LevelButtons/> successfully renders the component, matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="levelButton"
  >
    <button
      class="levelButtons"
      id="levelButton-7"
    >
      <span>
        Go to level 7
      </span>
    </button>
  </div>
</DocumentFragment>
`;
